mlflow/sklearn/__init__.py

```diff
@@ -825,6 +829,129 @@ def _get_metric_name_list():
     return metric_list
 
 
+def _patch_estimator_method_if_available(flavor_name, class_def, func_name, patched_fn, manage_run):
+    if not hasattr(class_def, func_name):
+        return
+
+    original = gorilla.get_original_attribute(
+        class_def, func_name, bypass_descriptor_protocol=False
+    )
+    # Retrieve raw attribute while bypassing the descriptor protocol
+    raw_original_obj = gorilla.get_original_attribute(
+        class_def, func_name, bypass_descriptor_protocol=True
+    )
+
+    if isinstance(raw_original_obj, property):
+
+        def real_original(self, *args, **kwargs):
+            bound_delegate_method = raw_original_obj.fget(self)
+            return bound_delegate_method(*args, **kwargs)
+
+        safe_patch(
+            flavor_name,
+            class_def,
+            func_name,
+            patched_fn,
+            manage_run=manage_run,
+            original_fn=real_original,
+        )
```

Comment:
Instead of adding an `original_fn` attribute to `safe_patch`, which may end up being abused by other autologging implementations, can we add this logic for handling property methods to the `safe_patch` implementation? Can we also add thorough tests for it in https://github.com/mlflow/mlflow/blob/master/tests/autologging/test_autologging_safety_unit.py?